django/contrib/staticfiles/storage.py

```diff
@@ -49,6 +49,9 @@ class HashedFilesMixin:
             r"""(url\(['"]{0,1}\s*(.*?)["']{0,1}\))""",
             (r"""(@import\s*["']\s*(.*?)["'])""", """@import url("%s")"""),
         )),
+        ("*.js", (
+            (r"(?m)^(//# (?-i:sourceMappingURL)=(.*))$", "//# sourceMappingURL=%s"),
+        )),
```

Comment:
this regex uses an `(?-i:...)` group to disable the case-insensitive flag around the `sourceMappingURL`. It's probably a bit defensive, but ideally we'd not force the case-insensitive flag on the patterns in `__init__`. But I can't think of a way of doing that in a bakcwards-compatible way.